### PR TITLE
[encoding] Add tests for GBK and gb18030 encoding

### PR DESCRIPTION
**Pull request by achristensen07**
These tests come from https://bugs.webkit.org/show_bug.cgi?id=218380
All tests pass in Firefox and in WebKit after that change.
Chrome has the following deviations from the standard:
1. The URL GBK encoding of the Yen sign (U+00A5) is %A3%A4 instead of %26%23165%3B
2. Decoding 0x80 with gb18030 results in a replacement character instead of the Euro sign (U+20A0)
3. The GBK decoder does not use the gb18030 decoder, which makes it less permissive.

### CODE REVIEW

File: encoding/legacy-mb-schinese/gb18030/gb18030-decoder.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<!doctype html>
```
Comment (inexorabletash) [inline encoding/legacy-mb-schinese/gb18030/gb18030-decoder.html:1]: Maybe make this a `.any.js` test to reduce boilerplate and increase coverage to workers?

File: encoding/legacy-mb-schinese/gb18030/gb18030-encoder.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
+<!doctype html>
+<meta charset=gb18030>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<script src=resources/ranges.js></script>
+<div id=log></div>
```
Comment (inexorabletash) [inline encoding/legacy-mb-schinese/gb18030/gb18030-encoder.html:6]: Is this `div` necessary these days? I thought testharness created it on the fly, but there may be other considerations.

**Review by inexorabletash — COMMENTED** [review]
Thanks!

File: encoding/legacy-mb-schinese/gb18030/gb18030-decoder.html (hunk 1 above)
Comment (achristensen07) [reply encoding/legacy-mb-schinese/gb18030/gb18030-decoder.html:1]: This test matches the pattern established by many other tests in the legacy-mb-* directories of not testing in workers and non-workers.  There are other tests that verify that encodings in worker contexts behave as they ought. There are already over a hundred thousand tests in these directories, and they take too long in debug builds that we don't even run them in debug builds in our continuous integration.  I'd rather not break with the pattern established by other multibyte encoding tests and add worker tests here unless another reviewer also thinks we should.

File: encoding/legacy-mb-schinese/gb18030/gb18030-encoder.html (hunk 2 above)
Comment (achristensen07) [reply encoding/legacy-mb-schinese/gb18030/gb18030-encoder.html:6]: It's not necessary but it was there before.  I'll remove it.

File: encoding/legacy-mb-schinese/gb18030/gb18030-decoder.html (hunk 1 above)
Comment (inexorabletash) [reply encoding/legacy-mb-schinese/gb18030/gb18030-decoder.html:1]: Fair points re: consistency and performance. (We could bulk convert files to `.window.js` but that's not in scope for this PR)

**Review by inexorabletash — APPROVED** [review]

**Comment by achristensen07** [comment]
I added another file since inexorabletash's approval.  It includes 10 code points that are decoded with GBK differently in Chromium.  They are these pointer values:
23834, 23835, 23836, 23837, 23838, 23840, 23841, 23842, 23843, 23844
which decode to these code points in Firefox, WebKit after adopting the spec, and the spec:
U+2e84, U+3473, U+3447, U+2e88, U+2e8b, U+359e, U+361a, U+360e, U+2e8c, U+2e97
but these code points in Chromium:
U+e819, U+e81a, U+e81b, U+e81c, U+e81d, U+e81f, U+e820, U+e821, U+e822, U+e823

**Comment by annevk** [comment]
Thanks for writing these. There was also #20361 but the i18n WG never got around to finishing those and cleaning them up seemed like quite a bit of effort.

**Review by annevk — APPROVED** [review]

**Review by inexorabletash — APPROVED** [review]
Sticking with "Approve" but a last minute observation that many variables aren't declared/scoped (with `var`, `let` or `const`) in many of these files/functions. Probably copy/paste from existing tests so not a big deal to fix, but I wouldn't object if they got a sprinkle of keywords.

**Comment by annevk** [comment]
I have rebased this since a bot seemed to be stuck and I also added a commit to clean up the const/let/semicolon stuff since I think I was responsible for the original bad style here.

**Pull request merged**